app/common/src/forms/TextInput.js

```diff
@@ -15,12 +18,16 @@ export const styles = {
     height: 40,
     backgroundColor: '#fff',
     marginBottom: 5,
+    flexDirection: 'row',
   },
 };
 
 export default function TextInput(props: Object) {
   return (
     <View style={styles.wrapper}>
+      {props.icon && (
+        <MaterialIcons {...defaultsDeep(props.icon, defaultIconProps)} />
```

Comment:
Now the question is: how to use different icon than from `MaterialIcons` package? 🙂 Because don't forget that you are modifying general purpose `TextInput`. There are few possibilities:

1. you cannot - meh 😖 
2. you are benevolent (`<TextInput icon={<Icon ... />}>`)
3. you introduce unified icon component and you'll detect the appropriate class by icon name

Reply:
With option 2 or 3 - where would you put `defaultIconProps`? These dimensions are specific for the use inside the TextInput, I don't think they belong to a general icon component.

I am OK with the option 1 for now, I would even simplify it to pass only the icon name (`<TextInput icon="location-city" />`), until we have the need for something else than grey Material icon. To switch from this simple approach to a more complicated approach will be super easy (and maybe YAGNI).

If we try to prepare for the unknown future, there is a risk that we will either overcomplicate it, or we won't address the future needs, or both.

Reply:
Example of a specialized component from the shared package: https://github.com/kiwicom/react-native-app/blob/master/app/core/src/components/relay/PrivateApiRenderer.js (you don't want Redux in common but you want to reuse the logic - split it).

`<TextInput icon="location-city" />` is OK for me. Better than pure config object with unclear structure.